server/accounts.go

```diff
@@ -274,7 +274,42 @@ func (a *Account) updateRemoteServer(m *AccountNumConns) {
 	a.strack[m.Server.ID] = sconns{conns: int32(m.Conns), leafs: int32(m.LeafNodes)}
 	a.nrclients += int32(m.Conns) - prev.conns
 	a.nrleafs += int32(m.LeafNodes) - prev.leafs
+
+	mtce := a.mconns != jwt.NoLimit && (len(a.clients)-int(a.sysclients)+int(a.nrclients) > int(a.mconns))
+	// If we are over here some have snuck in and we need to rebalance.
+	// All others will probably be doing the same thing but better to be
+	// conservative and bit harsh here. Clients will reconnect if we over compensate.
+	var clients []*client
+	if mtce {
+		clients = make([]*client, 0, len(a.clients))
+		for c := range a.clients {
+			clients = append(clients, c)
+		}
+		sort.Slice(clients, func(i, j int) bool {
+			return clients[i].start.After(clients[j].start)
```

Comment:
`start` should be set only on CONNECT, but we access without locking here..

Reply:
Yes understood. If we see a race we can take a look but I think its better than locking each client.